operator/pkg/client/gvkn.go

```diff
@@ -0,0 +1,30 @@
+package client
+
+import (
+	"crypto/sha256"
+	"fmt"
+
+	"gopkg.in/yaml.v2"
+)
+
+type OverlySimpleGVKWithName struct {
+	APIVersion string               `yaml:"apiVersion"`
+	Kind       string               `yaml:"kind"`
+	Metadata   OverlySimpleMetadata `yaml:"metadata"`
+}
+
+type OverlySimpleMetadata struct {
+	Name string `yaml:"name"`
+}
+
+func GetGVKWithName(content []byte) string {
+	o := OverlySimpleGVKWithName{}
+
+	if err := yaml.Unmarshal(content, &o); err != nil {
+		return ""
+	}
+
+	h := sha256.New()
+	h.Write([]byte(fmt.Sprintf("%s-%s-%s", o.APIVersion, o.Kind, o.Metadata.Name)))
```

Comment:
should we include namespace here too?
since IIRC you can have things with the same GVK+N if the namespaces are different

Reply:
And people probably want to remove the old version when changing namespaces

Reply:
"people will never manually specify the namespace" is also an acceptable resolution method of course